### PR TITLE
Generate modules metadata

### PR DESCRIPTION
**Pull request by bhsubra**
Changes in the PR include:
1. Generate modules metadata that'll be used in completion scenarios
2. I'll have a follow up PR to write the generated data to storage account container

Verified the changes in dogfood via:
https://github.com/bhsubra/bicep-registry-modules/pull/1

### CODE REVIEW

File: scripts/github-actions/get-modules-metadata.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+async function getModulesMetadata({ require, core }) {
```
Comment (shenglol) [inline scripts/github-actions/get-modules-metadata.js:1]: Mind adding a [JSDoc ](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) comment to annotate the types of parameters?

File: scripts/github-actions/get-sub-directory-names.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @param {typeof import("fs")} fs
+ * @param {string} dir
+ */
+function getSubdirNames(dir) {
+  const fs = require("fs");
```
Comment (shenglol) [inline scripts/github-actions/get-sub-directory-names.js:6]: Should `fs` remain a parameter and be passed from the caller?

File: scripts/github-actions/get-modules-metadata.js (hunk 1 above)
Comment (bhsubra) [reply scripts/github-actions/get-modules-metadata.js]: Sure, updated here:
https://github.com/Azure/bicep-registry-modules/commit/20d412d9da3b54faf759004a8783918aee430fe4

Thanks!

File: scripts/github-actions/get-sub-directory-names.js (hunk 2 above)
Comment (bhsubra) [reply scripts/github-actions/get-sub-directory-names.js]: Updated here:
https://github.com/Azure/bicep-registry-modules/commit/20d412d9da3b54faf759004a8783918aee430fe4

Thanks!

File: .github/workflows/get-modules-metadata.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        required: true
+      TENANT_ID:
+        required: true
+      SUBSCRIPTION_ID:
+        required: true
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
```
Comment (shenglol) [inline .github/workflows/get-modules-metadata.yml:16]: Are the secrets needed? If not, we should remove them.

Comment (bhsubra) [reply .github/workflows/get-modules-metadata.yml:16]: Yes, I'll be sending out a follow up PR to login to azure and write the modules metadata to storage account container. Let me know if you want me to remove it here and include it in other PR.

Comment (shenglol) [reply .github/workflows/get-modules-metadata.yml:16]: No, in that case we can leave them as is.

**Review by shenglol — APPROVED** [review]

**Pull request merged**